src/tests/system/tests/test_pysss_nss_idmap.py

```diff
@@ -0,0 +1,345 @@
+"""
+SSSD Python Client identity Lookups
+
+:requirement: Python binding of libsss_nss_idmap.so
+"""
+
+from __future__ import annotations
+
+import ast
+
+import pytest
+from sssd_test_framework.roles.ad import AD
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.roles.samba import Samba
+from sssd_test_framework.topology import KnownTopology, KnownTopologyGroup
+
+
+def get_sid_name(provider: GenericProvider):
+    if isinstance(provider, Samba):
+        return "objectSid"
+    elif isinstance(provider, AD):
+        return "SID"
+    elif isinstance(provider, IPA):
+        return "ipaNTSecurityIdentifier"
+    return None
+
+
+def user_setup(provider: GenericProvider, client: Client):
+    sid_name = get_sid_name(provider)
+    assert sid_name is not None
+
+    user = provider.user("user1").add()
+    user_sid = user.get([sid_name])[sid_name][0]
+
+    client.sssd.restart()
+
+    result = client.tools.getent.passwd(user.name)
+    assert result is not None, f"'{user.name}' missing"
+    assert result.name == user.name, f"'{user.name}' has wrong name {result.name}"
+    user_id = result.uid
+
+    return (user, user_id, user_sid)
+
+
+def group_setup(provider: GenericProvider, client: Client):
+    sid_name = get_sid_name(provider)
+    assert sid_name is not None
+
+    group = provider.group("group1").add()
+    group_sid = group.get([sid_name])[sid_name][0]
+
+    client.sssd.restart()
+
+    result = client.tools.getent.group(group.name)
+    assert result is not None, f"'{group.name}' missing"
+    assert result.name == group.name, f"'{group.name}' has wrong name {result.name}"
+    group_id = result.gid
+
+    return (group, group_id, group_sid)
+
+
+def run_pysss_nss_idmap(client: Client, command: str, input):
+    if isinstance(input, str):
+        input = f'"{input}"'
+
+    return client.host.conn.run(f"""python -c 'import pysss_nss_idmap; print(pysss_nss_idmap.{command}({input}))'""")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyDC)
+def test_user_by_name(client: Client, provider: GenericProvider):
+    """
+    :title: Lookup the SID of a user by name
```

Comment:
We started generalizing the doc string to provide more of an overview, and the specific details are in the code. Something like;

```
:setup:
    1. Add user and start sssd
:steps:
    1. Lookup user by name and username
    2. Lookup user by group name
:expectedresults:
    1. User found
    2. User not found
```

Reply:
I updated the doc strings in the latest version according to your suggestions.

Reply:
So much better, thank you so much.